classes/plugininfo/paymentgateway.php

```diff
@@ -15,18 +15,43 @@
 // along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
 
 /**
- * Subplugin info class.
+ * Defines the subplugin type 'paymentgateway'
+ *
+ * File         paymentgateway.php
+ * Encoding     UTF-8
  *
  * @package     tool_paymentplugin
- * @author      Haruki Nakagawa
+ *
  * @copyright   MAHQ
  * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
- */
+ **/
 
 namespace tool_paymentplugin\plugininfo;
 
 defined('MOODLE_INTERNAL') || die();
 
-class paymentgateway extends \core\plugininfo\base {
+// @See https://docs.moodle.org/dev/Subplugins#Settings_pages
+class paymentgateway extends \core\plugininfo\base  {
+
+    // @credit https://github.com/catalyst/moodle-tool_mfa/blob/master/classes/plugininfo/factor.php
+    public function load_settings(\part_of_admin_tree $adminroot, $parentnodename, $hassiteconfig) {
```

Comment:
This was pulled right from mfa, so it should be good to stay. All the things I removed were my half-complete attempts at loading the settings. If you think it truly doesn't have to be there, then I'll remove it.

Reply:
I think you should copy over the `get_settings_section_name()` method from the MFA plugininfo file too, as it's used in line 47.  I did end up implementing both functions in paypal_settings branch so that Quyen could work, so it will be pushed to dev once Quyen finishes paypal settings anyway.